### PR TITLE
Brush bug

### PR DESCRIPTION
**Pull request by veev**
Fixed positioning of brush in relation to graph with magic number `(0.03 * this.props.width)`
Updated axis scales on resize
Updated brush extent on resize

I believe the brush issues should be fixed by this PR. Please test and let me know what you think.

### CODE REVIEW

**Comment by lxu213** [comment]
Just got back from vacation, will review tomorrow!

File: src/components/Graph/MainGraph.js [hunk 1]
```diff
@@ -443,7 +444,7 @@ class MainGraph extends Component {
                             height={80}
                             series={this.state.allDatesSeries}
                             dates={this.state.dates}
-                            x={margin.yAxis}
+                            x={margin.yAxis + (this.props.width * 0.03)}
```
Comment (lxu213) [inline src/components/Graph/MainGraph.js:447]: How did you derive this `0.03` number? If it's constant, could we move it to the `constants.js` file with a comment on how you calculated this value?

**Comment by lxu213** [comment]
@veev One thing I'm noticing is that if I resize the window (increasing it) and then try to drag the right margin of the Brush, the Graph doesn't update until I drag to a certain point. I'm wondering if this is when a certain threshold gets crossed? 

Screen Shot 2020-07-05 at 11 30 38 AM
Screen Shot 2020-07-05 at 11 30 52 AM

Doesn't shift until: 
Screen Shot 2020-07-05 at 11 31 04 AM

File: src/components/Graph/MainGraph.js (hunk 1 above)
Comment (veev) [reply src/components/Graph/MainGraph.js:447]: Sure thing. The issue of why the brush area wasn't lining up with the graph using just margin.yAxis has to do with the .graph-wrapper div being set to flex, and the y-axis label being rendered from a div not svg. It is difficult to calculate how much space will be allotted for this before rendering graph, so hard to pass in to the component before Graph itself mounts. The 0.03 constant was derived from a few tests at different screen widths.

**Comment by veev** [comment]
Hmm, not sure why a threshold would be an issue. It's set to update on a change in the prop.width or prop.height. Updating should be calling the extent again, but if for some reason this isn't firing that could be the case. Let me see if I can repro this behavior.

**Comment by veev** [comment]
Ah I think I see what you're talking about, sorry was interpreting a Brush update problem instead of Graph. Will look into why this is happening.

**Comment by veev** [comment]
I think this might be because we are not calling this.update in handleBrushRange, we are just setting dateRange to state. GraphContainer and Graph needs new dates to have its props update and change the xScale. I can refactor this so they are just looking for dateRange, or we can call this.update within handleBrushChange. Preferences on which approach to take?

**Comment by veev** [comment]
Also, should I refactor all of the width multipliers in MainContainer and put in the constants folder?

**Comment by lxu213** [comment]
By "I can refactor this so they are just looking for dateRange", are you
talking about componentDidUpdate in Graph and GraphContainer?

I think that first strategy might be cleaner than calling this.update in
the brush handler just because there's a lot of other stuff going on in
update.

And yes, I think moving width multipliers to a constants file is a good
idea.

**Comment by veev** [comment]
By "I can refactor this so they are just looking for dateRange", are you
talking about componentDidUpdate in Graph and GraphContainer?

Yes, referring to how componentDidUpdate works in GraphContainer to re-calculate scales, which are passed into Graph. However, it appears that Graph needs an updated selectedDates prop to re-render correctly, so the first approach is not enough. I will have to call this.update in the brush handler after all.

I moved width multipliers and will add this.update and push so you can review.

**Comment by veev** [comment]
Ok, I was wrong about the cause of the issue. It has nothing to do with not calling this.update ( I was looking at handleBrushStart by accident). It has to do with the fact that the range that the brush is returning in the brush handler are wildly different than what is being displayed on the axes. It's like when the window size got increased, the xScale updated, but the original brush xScale for the handler did not. I'm looking into a fix, but just wanted to update you in case you looked at the code and were confused.

**Comment by veev** [comment]
Ok this issue should now be resolved. I have also refactored the width dimensions multipliers into the constants.js file.

**Pull request merged**